### PR TITLE
US127820 - no html templates available

### PR DESCRIPTION
**Pull request by coffeexcode**
Co-authored with @quinnbudan 

## Relevant Rally Links

- [US127820](https://rally1.rallydev.com/#/289692574792d/custom/355050439968?detail=%2Fuserstory%2F602981068972&fdp=true?fdp=true): FACE HTML file - Support Template Selection - Handle the case when the API returns 0 templates

## Description

This PR handles the case when no HTML templates are returned.

## Goal/Solution

This PR handles the case when no HTML templates are returned. The solution is to show a small message in the dropdown and support screen readers.

## Video/Screenshots

N/A

## Remaining Work
- [x] Dev Testing: another developer will test this code on their machine

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js [hunk 1]
```diff
@@ -34,14 +36,22 @@ class ContentFileDetail extends AsyncContainerMixin(SkeletonMixin(ErrorHandlingM
...
 				.d2l-page-content-label-select-template-container {
 					align-items: center;
 					display: flex;
 					justify-content: space-between;
 					margin-bottom: 6px;
 				}
-			`
+				.d2l-new-html-editor-container {
+					margin-top: 6px;
+				}
```
Comment (prajapac) [inline components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js:50]: I believe you don't need this still.

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js [hunk 2]
```diff
@@ -75,7 +85,7 @@ class ContentFileDetail extends AsyncContainerMixin(SkeletonMixin(ErrorHandlingM
 
 			switch (contentFileEntity.fileType) {
 				case FILE_TYPES.html:
-					pageRenderer = this._renderHtmlEditor(pageContent);
+					pageRenderer = this._renderHtmlEditor(pageContent, this.htmlTemplatesHref);
```
Comment (prajapac) [inline components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js:88]: `this.htmlTemplatesHref` is global, so no need to pass around.

Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js:88]: It got left in from the rebase... will remove

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js (hunk 1 above)
Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js:50]: It got left in from the rebase... will remove

Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js]: resolved in a060253

**Review by prajapac — DISMISSED** [review]
LGTM

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js (hunk 2 above)
Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js]: resolved in a060253

File: components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js [hunk 3]
```diff
@@ -235,6 +225,43 @@ class ContentFileDetail extends AsyncContainerMixin(SkeletonMixin(ErrorHandlingM
 			</div>`;
 	}
 
+	_renderHtmlTemplates() {
+		if (this.htmlFileTemplates.length === 0) {
```
Comment (ianlaidlaw) [inline components/d2l-activity-editor/d2l-activity-content-editor/content-file/d2l-activity-content-file-detail.js:229]: super duper nit, no need to change but you may prefer:
```
this.htmlFileTemplates.length.length === 0
```
is the same as 
```
!this.htmlFileTemplates.length
```

**Review by ianlaidlaw — APPROVED** [review]
LGTM, also dev tested

**Review by CodeBaboon — APPROVED** [review]
Qualm free

**Pull request merged**